### ISSUE TITLE
Arch Linux AUR package

### ISSUE THREAD

**Issue by EVV1E** [nik012003/ripdrag#18]
I submitted ripdrag as an AUR package and plan on maintaining it in the future.

If anybody has recommendations on how to improve the PKGBUILD please notify me! :)

https://aur.archlinux.org/packages/ripdrag-git

edit: changed to new url

**Comment by RoryNesbitt** [comment]
@Fabillotic the aur package fails to install at the build stage, but if you remove `--frozen` from the cargo command it will work. I don't know if `--frozen` is otherwise useful but removing it worked for me

**Comment by Fabillotic** [comment]
@RoryNesbitt 
that's weird...
Does it work when adding this function to the PKGBUILD?
`prepare() {
    export RUSTUP_TOOLCHAIN=stable
    cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
}`

**Comment by RoryNesbitt** [comment]
Now that it's been installed the regular PKGBUILD works. I'll update later when I'm on my laptop

**Comment by Fabillotic** [comment]
Seems like an issue with cargo needing to download dependencies

**Comment by RoryNesbitt** [comment]
yes, I don't have the exact error now but it was to do with installing clap as a dependency

**Comment by Fabillotic** [comment]
should now work with ripdrag 0.2.1-2

**Comment by RoryNesbitt** [comment]
tried now that I'm on my laptop and yeah, installed without issue

**Comment by pbosab** [comment]
> should now work with ripdrag 0.2.1-2

you should remove ripdrag -V from

```
check() {
	cd "$pkgname-$pkgver"
	./target/release/ripdrag -V
}
```

otherwise you cannot update it if not in a graphical environment.

```
==> Starting check()...

(ripdrag:6189): Gdk-WARNING **: 10:36:42.854: No such backend: *

(ripdrag:6189): Gtk-WARNING **: 10:36:42.854: Failed to open display
==> ERROR: A failure occurred in check().
    Aborting...
 -> error making: ripdrag-exit status 4
 -> Failed to install the following packages. Manual intervention is required:
ripdrag - exit status 4

```

@nik012003 issue? or to-do?

```

$ ripdrag --help

(ripdrag:13276): Gdk-WARNING **: 10:43:05.957: No such backend: *

(ripdrag:13276): Gtk-WARNING **: 10:43:05.957: Failed to open display
```

**Comment by nik012003** [comment]
Oops, my bad. Should be an easy fix. Will fix ASAP.

**Comment by nik012003** [comment]
> > should now work with ripdrag 0.2.1-2
> 
> you should remove ripdrag -V from
> 
> ```
> check() {
> 	cd "$pkgname-$pkgver"
> 	./target/release/ripdrag -V
> }
> ```
> 
> otherwise you cannot update it if not in a graphical environment.
> 
> ```
> ==> Starting check()...
> 
> (ripdrag:6189): Gdk-WARNING **: 10:36:42.854: No such backend: *
> 
> (ripdrag:6189): Gtk-WARNING **: 10:36:42.854: Failed to open display
> ==> ERROR: A failure occurred in check().
>     Aborting...
>  -> error making: ripdrag-exit status 4
>  -> Failed to install the following packages. Manual intervention is required:
> ripdrag - exit status 4
> ```
> 
> @nik012003 issue? or to-do?
> 
> ```
> 
> $ ripdrag --help
> 
> (ripdrag:13276): Gdk-WARNING **: 10:43:05.957: No such backend: *
> 
> (ripdrag:13276): Gtk-WARNING **: 10:43:05.957: Failed to open display
> ```

Thanks, fixed in v0.3.2

**Comment by pbosab** [comment]
> Thanks, fixed in v0.3.2

That was fast, thanks.

@Fabillotic so going back to my first reply: removing the ripdrag -V is not needed anymore... anyway i'm not a developer, but i think it's better to make a -git pkgbuild instead of one based on releases in an early development stage, or u'll end up editing it everytime and getting marked out of date everyday.

**Comment by Fabillotic** [comment]
Updated to v0.3.2, looking to make -git tracking package instead

**Comment by Fabillotic** [comment]
# Important update

**The new AUR package is now `ripdrag-git` instead of `ripdrag`**

I have made a request to merge the old package into the new one.
As the name implies, ripdrag-git will now track this Github repository.

**Comment by Fabillotic** [comment]
@pbosab thank's for the advice

**Comment by Deshdeepak1** [comment]
Please provide a pre- compiled binary.

**Comment by Fabillotic** [comment]
> Please provide a pre- compiled binary. 

Sorry but I agree with @pbosab on this way of packaging. A -git tracking package is more appropriate at this stage of development. It also shouldn't be an issue building the package although it may take some time. If you have any issues building the package, please let me know.
